xdsl/backend/arm_neon/lowering/convert_arith_to_neon.py

```diff
@@ -0,0 +1,40 @@
+from xdsl.dialects.arm_neon import NeonArrangement
+from xdsl.dialects.builtin import Float16Type, Float32Type, Float64Type, VectorType
+
+
+def get_arrangement_from_vec_type(vec_type: VectorType) -> NeonArrangement:
```

Comment:
Could this be a dictionary mapping things like `vector<8xf16>` to `.H` and so on? We can then raise a single exception saying something like `Invalid vector type for ARM NEON: vector<17xi5>`.

Reply:
Feels like it could also be defined on the NeonArrangement class itself as a static method.

Reply:
I made these updates, but I think I will leave it at that for this PR and open a new one for the next changes, since plans have changed a bit since I started this